core/src/stages/move_to.cpp

```diff
@@ -238,7 +251,35 @@ bool MoveTo::compute(const InterfaceState& state, planning_scene::PlanningSceneP
 		add_frame(ik_pose_world, "ik frame");
 
 		// transform target pose such that ik frame will reach there if link does
-		target = target * ik_pose_world.inverse() * scene->getCurrentState().getGlobalLinkTransform(link);
+		const auto ik_pose_link = ik_pose_world.inverse() * scene->getCurrentState().getGlobalLinkTransform(link);
+		target = target * ik_pose_link;
+
+		// Add circular arc path constraints, if enabled.
+		// Note that this replaces all other path constraints.
+		if (!props.get("arc_constraint").empty()) {
+			RCLCPP_INFO(LOGGER, "Adding circular arc constraint.");
+			auto [arc_constraint_type, arc_constraint_pose] =
+			    props.get<std::pair<std::string, geometry_msgs::msg::PoseStamped>>("arc_constraint");
+
+			Eigen::Isometry3d arc_target;
+			tf2::fromMsg(arc_constraint_pose.pose, arc_target);
+			arc_target = scene->getFrameTransform(arc_constraint_pose.header.frame_id) * arc_target;
+			add_frame(arc_target, "constraint frame");
+			arc_target = arc_target * ik_pose_link;
+			geometry_msgs::msg::PoseStamped arc_target_pose;
+			arc_target_pose.header.frame_id = scene->getPlanningFrame();
+			arc_target_pose.pose = Eigen::toMsg(arc_target);
+
+			path_constraints.name = arc_constraint_type;
+			moveit_msgs::msg::PositionConstraint pos_constraint;
+			pos_constraint.header.frame_id = arc_target_pose.header.frame_id;
+			pos_constraint.link_name = link->getName();
+			pos_constraint.constraint_region.primitive_poses.resize(1);
+			pos_constraint.constraint_region.primitive_poses[0] = arc_target_pose.pose;
+			pos_constraint.weight = 1.0;  // Shouldn't matter as it's the only constraint
+			path_constraints.position_constraints.resize(1);
+			path_constraints.position_constraints[0] = pos_constraint;
+		}
```

Comment:
This is the clunkiest part of the interface, but I couldn't come up with something better... Thoughts?